src/utils.re

```diff
@@ -0,0 +1,158 @@
+let kebabToCamel =
+  String.foldi
+    init::""
+    f::(
+      fun _ accum char =>
+        if (accum == "") {
+          Char.to_string char
+        } else if (accum.[String.length accum - 1] == '-') {
+          String.slice accum 0 (-1) ^ (Char.to_string char |> String.capitalize)
+        } else {
+          accum ^ Char.to_string char
+        }
+    );
+
+let nonBlank s =>
+  switch (String.strip s) {
+  | "" => false
+  | _ => true
+  };
+
+/* assumes there is a suffix to chop. Throws otherwise */
+let chopSuffixExn str => String.slice str 0 (String.rindex_exn str '.');
+
+/* jenga helpers */
+let bash command => Action.process dir::Path.the_root prog::"bash" args::["-c", command] ();
+
+let bashf fmt => ksprintf bash fmt;
+
+let relD dir::dir str => Dep.path (Path.relative dir::dir str);
+
+let bindD f dep => Dep.bind dep f;
+
+let mapD f dep => Dep.map dep f;
+
+let rel = Path.relative;
+
+
+/** Path helpers **/
+let fileNameNoExtNoDir path => Path.basename path |> chopSuffixExn;
+
+let isInterface path => {
+  let base = Path.basename path;
+  String.is_suffix base suffix::".rei" || String.is_suffix base suffix::".mli"
+};
+
+let hasInterface sourcePaths::sourcePaths path =>
+  not (isInterface path) &&
+  List.exists
+    sourcePaths
+    f::(fun path' => isInterface path' && fileNameNoExtNoDir path' == fileNameNoExtNoDir path);
+
+let nodeModulesRoot = Path.relative dir::Path.the_root "node_modules";
+
+let buildDirRoot = Path.relative dir::Path.the_root "_build";
+
+let topSrcDir = Path.relative dir::Path.the_root "src";
+
+
+/** this rebel-specific helpers **/
+type moduleName =
+  | Mod string;
+
+type libName =
+  | Lib string;
+
+let tsp = Path.to_string;
+
+let tsm (Mod s) => s;
+
+let tsl (Lib s) => s;
+
+let libToModule (Lib name) => Mod (String.capitalize name |> kebabToCamel);
+
+let pathToModule path => Mod (fileNameNoExtNoDir path |> String.capitalize);
+
+let namespacedName libName::libName path::path =>
+  tsm (libToModule libName) ^ "__" ^ tsm (pathToModule path);
+
+/* FIXME Remove after bloomberg/bucklescript#757 is fixed */
+let bsNamespacedName libName::libName path::path => {
+  let bsKebabToCamel =
+    String.foldi
+      init::""
+      f::(
+        fun _ accum char =>
+          if (accum == "") {
+            Char.to_string char |> String.lowercase
+          } else if (
+            accum.[String.length accum - 1] == '-'
+          ) {
+            String.slice accum 0 (-1) ^ (Char.to_string char |> String.capitalize)
+          } else {
+            accum ^ Char.to_string char
+          }
+      );
+  let bsLibToModule (Lib name) => Mod (String.capitalize name |> bsKebabToCamel);
+  tsm (bsLibToModule libName) ^ "__" ^ tsm (pathToModule path)
+};
+
+/* let topLibName = {
+  let packageJsonPath = Path.relative dir::Path.the_root "package.json";
+  from_file (Path.to_string packageJsonPath) |> Util.member "name" |> Util.to_string |> (
+    fun name => Lib name
+  )
+}; */
+let topLibName = Lib "src";
```

Comment:
Yeah. All the top level artificats will go to `_build/src`  in this implemenation.